crates/bevy_core_pipeline/src/oit/resolve/mod.rs

```diff
@@ -0,0 +1,213 @@
+#[derive(Resource)]
+pub struct OitResolvePipeline {
+    /// View bind group layout
+    pub view_bind_group_layout: BindGroupLayout,
+    /// Depth bind group layout
+    pub oit_depth_bind_group_layout: BindGroupLayout,
+}
+
+impl FromWorld for OitResolvePipeline {
+    fn from_world(world: &mut World) -> Self {
+        let render_device = world.resource::<RenderDevice>();
+
+        let view_bind_group_layout = render_device.create_bind_group_layout(
+            "oit_resolve_bind_group_layout",
+            &BindGroupLayoutEntries::sequential(
+                ShaderStages::FRAGMENT,
+                (
+                    uniform_buffer::<ViewUniform>(true),
+                    // layers
+                    storage_buffer_sized(false, None),
+                    // layer ids
+                    storage_buffer_sized(false, None),
+                ),
+            ),
+        );
+
+        let oit_depth_bind_group_layout = render_device.create_bind_group_layout(
+            "oit_depth_bind_group_layout",
+            &BindGroupLayoutEntries::single(ShaderStages::FRAGMENT, texture_depth_2d()),
+        );
+        OitResolvePipeline {
+            view_bind_group_layout,
+            oit_depth_bind_group_layout,
+        }
+    }
+}
+
+#[derive(Component, Deref, Clone, Copy)]
+pub struct OitResolvePipelineId(pub CachedRenderPipelineId);
+
+/// This key is used to cache the pipeline id and to specialize the render pipeline descriptor
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct OitResolvePipelineKey {
+    hdr: bool,
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn queue_oit_resolve_pipeline(
+    mut commands: Commands,
+    pipeline_cache: Res<PipelineCache>,
+    resolve_pipeline: Res<OitResolvePipeline>,
+    views: Query<(Entity, &ExtractedView), With<OrderIndependentTransparencySettings>>,
+    // Store the key with the id to make the clean up logic easier
+    // This also means it will always replace the entry if the key changes so nothing to clean up
+    mut cached_pipeline_id: Local<EntityHashMap<(OitResolvePipelineKey, CachedRenderPipelineId)>>,
+) {
+    let mut current_view_entities = EntityHashSet::default();
+    for (e, view) in &views {
+        current_view_entities.insert(e);
+        let key = OitResolvePipelineKey { hdr: view.hdr };
+
+        if let Some((cached_key, id)) = cached_pipeline_id.get(&e) {
+            if *cached_key == key {
+                commands.entity(e).insert(OitResolvePipelineId(*id));
+                continue;
+            }
+        }
+
+        let desc = specialize_oit_resolve_pipeline(key, &resolve_pipeline);
+
+        let pipeline_id = pipeline_cache.queue_render_pipeline(desc);
+        commands.entity(e).insert(OitResolvePipelineId(pipeline_id));
+        cached_pipeline_id.insert(e, (key, pipeline_id));
+    }
+
+    // Clear cache for views that don't exist anymore
+    // We can't rely on removal detection here because components in the render world aren't persisted
```

Comment:
They are now!

Reply:
Well, technically not _yet_, but I guess they will be soon enough that I should update that comment. I'm also not sure if that means we'll be able to rely on change detection for that.

Reply:
Oh I thought it was merged already 😅